pandas/tests/io/test_parquet.py

```diff
@@ -0,0 +1,232 @@
+        # non-default index
+        for index in [[2, 3, 4],
+                      pd.date_range('20130101', periods=3),
+                      list('abc'),
+                      [1, 3, 4],
+                      pd.MultiIndex.from_tuples([('a', 1), ('a', 2),
+                                                 ('b', 1)]),
+                      ]:
+
+            df.index = index
+            self.check_error_on_write(df, engine, ValueError)
+
+        # index with meta-data
+        df.index = [0, 1, 2]
+        df.index.name = 'foo'
+        self.check_error_on_write(df, engine, ValueError)
+
+        # column multi-index
+        df.index = [0, 1, 2]
+        df.columns = pd.MultiIndex.from_tuples([('a', 1), ('a', 2), ('b', 1)]),
+        self.check_error_on_write(df, engine, ValueError)
+
+    @pytest.mark.parametrize('compression', [None, 'gzip', 'snappy', 'brotli'])
+    def test_compression(self, engine, compression):
+
+        if compression == 'snappy':
+            try:
+                import snappy  # noqa
+            except ImportError:
+                pytest.skip("no snappy, skipping")
+
+        elif compression == 'brotli':
+            try:
+                import brotli  # noqa
+            except ImportError:
+                pytest.skip("no brotli, skipping")
+
+        df = pd.DataFrame({'A': [1, 2, 3]})
+        self.check_round_trip(df, engine, compression=compression)
+
+
+class TestParquetPyArrow(Base):
+
+    def test_basic(self, pa):
+
+        df = pd.DataFrame({'string': list('abc'),
+                           'int': list(range(1, 4)),
+                           'uint': np.arange(3, 6).astype('u1'),
+                           'float': np.arange(4.0, 7.0, dtype='float64'),
+                           'bool': [True, False, True],
+                           })
+
+        self.check_round_trip(df, pa)
+
+    def test_duplicate_columns(self, pa):
+
+        # not currently able to handle duplicate columns
+        df = pd.DataFrame(np.arange(12).reshape(4, 3),
+                          columns=list('aaa')).copy()
+        self.check_error_on_write(df, pa, ValueError)
+
+    def test_unsupported(self, pa):
+        from pyarrow.error import ArrowException
+
+        # period
+        df = pd.DataFrame({'a': pd.period_range('2013', freq='M', periods=3)})
+        self.check_error_on_write(df, pa, ArrowException)
+
+        # categorical
+        df = pd.DataFrame({'a': pd.Categorical(list('abc'))})
+        self.check_error_on_write(df, pa, ArrowException)
+
+        # date_range
+        df = pd.DataFrame({'a': pd.date_range('20130101', periods=3)})
+        self.check_error_on_write(df, pa, ArrowException)
+
+        # date_range w/tz
+        df = pd.DataFrame({'a': pd.date_range('20130101', periods=3,
+                                              tz='US/Eastern')})
+        self.check_error_on_write(df, pa, ArrowException)
+
+    def test_mixed(self, pa):
+        # mixed python objects are returned as None ATM
```

Comment:
I would raise as this is not serializable. (could certainly do this at pandas level, but I think this is something that the impl library should do). a mixed-dtype column needs to have *something* happen to it (via encoding or otherwise). But these are decisions that cannot be made automatically. IIRC fastparquet has the option to make these into JSON string. Not suggesting anything, just error checking / NotImplementedError for now.